api_tests/nodes/views/test_node_embeds.py

```diff
@@ -1,100 +1,129 @@
+    def contrib_two(self):
+        return AuthUserFactory()
+
+    @pytest.fixture()
+    def contribs(self, contrib_one, contrib_two):
+        return [contrib_one, contrib_two]
+
+    @pytest.fixture()
+    def auth(self, user):
+        return Auth(user)
+
+    @pytest.fixture()
+    def make_public_node(self, user):
+        return functools.partial(ProjectFactory, is_public=False, creator=user)
+
+    @pytest.fixture()
+    def root_node(self, auth, contrib_one, contrib_two, make_public_node):
+        root_node = make_public_node()
+        root_node.add_contributor(contrib_one, ['read', 'write'], auth=auth, save=True)
+        root_node.add_contributor(contrib_two, ['read', 'write'], auth=auth, save=True)
+        return root_node
 
-    def setUp(self):
-        super(TestNodeEmbeds, self).setUp()
+    @pytest.fixture()
+    def child_one(self, auth, contrib_one, contrib_two, make_public_node, root_node):
+        child_one = make_public_node(parent=root_node)
+        child_one.add_contributor(contrib_one, ['read', 'write'], auth=auth, save=True)
+        child_one.add_contributor(contrib_two, ['read', 'write'], auth=auth, save=True)
+        return child_one
 
-        self.user = AuthUserFactory()
-        self.auth = Auth(self.user)
-        make_public_node = functools.partial(ProjectFactory, is_public=False, creator=self.user)
-        self.root_node = make_public_node()
-        self.child1 = make_public_node(parent=self.root_node)
-        self.child2 = make_public_node(parent=self.root_node)
+    @pytest.fixture()
+    def child_two(self, make_public_node, root_node):
+        return make_public_node(parent=root_node)
 
-        self.contribs = [AuthUserFactory() for i in range(2)]
-        for contrib in self.contribs:
-            self.root_node.add_contributor(contrib, ['read', 'write'], auth=self.auth, save=True)
-            self.child1.add_contributor(contrib, ['read', 'write'], auth=self.auth, save=True)
+    @pytest.fixture()
+    def subchild(self, child_two, contrib_one):
+        return ProjectFactory(parent=child_two, creator=contrib_one)
 
-        self.contrib1 = self.contribs[0]
-        self.contrib2 = self.contribs[1]
-        self.subchild = ProjectFactory(parent=self.child2, creator=self.contrib1)
+    def test_node_embeds(self, app, user, contrib_one, contribs, subchild, root_node, child_one, child_two):
 
-    def test_embed_children(self):
-        url = '/{0}nodes/{1}/?embed=children'.format(API_BASE, self.root_node._id)
+    #   test_embed_children
+        url = '/{0}nodes/{1}/?embed=children'.format(API_BASE, root_node._id)
 
-        res = self.app.get(url, auth=self.user.auth)
+        res = app.get(url, auth=user.auth)
         embeds = res.json['data']['embeds']
-        ids = [self.child1._id, self.child2._id]
+        ids = [child_one._id, child_two._id]
         for child in embeds['children']['data']:
-            assert_in(child['id'], ids)
+            assert child['id'] in ids
```

Comment:
@Nesiehr set protects against order being different:
<img width="205" alt="screen shot 2017-06-08 at 10 15 22 am" src="https://user-images.githubusercontent.com/1322421/26933065-6ca12ece-4c33-11e7-8b51-7f30c9184a0d.png">
Since the first version was iterating and checking `in` there is no assumptions about order being made, `set` keeps that assumption.
Also it needs to be set([child_one._id, child_two._id]) (nested in a list, whoops)